tests/php/src/Factories/OrderFactory.php

```diff
@@ -0,0 +1,225 @@
+<?php
+namespace WeDevs\Dokan\Test\Factories;
+
+use WC_Abstract_Order;
+use WC_Coupon;
+use WC_Order;
+use WC_Order_Item_Fee;
+use WC_Order_Item_Shipping;
+use WeDevs\Dokan\Test\Helpers\WC_Helper_Order;
+use WP_UnitTest_Factory_For_Thing;
+use WC_Order_Item;
+
+class OrderFactory extends WP_UnitTest_Factory_For_Thing {
+    /**
+     * @var array<WC_Order_Item>
+     */
+    protected $order_items = [];
+
+    /**
+     * @var array<WC_Coupon>
+     */
+    protected $coupon_items = [];
+
+    /**
+     * Factory Instance.
+     *
+     * @param DokanFactory $factory
+     */
+    public function __construct( $factory = null ) {
+        parent::__construct( $factory );
+        $this->default_generation_definitions = array(
+            'status'      => 'pending',
+            'customer_id' => 1,
```

Comment:
We should not set ` 'customer_id' => 1,` concretely. we should use customer factory here.

Reply:
If we use Customer factory here then always a customer will be created.  Customer factory should be used  when order factory is used.  It with create.
 ```
 $this->factory()->order->create(
    [
        'customer_id' => $this->factory()->customer->create(),
        'status' => 'complete'
    ]
 )
```